src/main/java/com/google/cloud/tools/managedcloudsdk/update/Updater.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright 2017 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.tools.managedcloudsdk.update;
+
+import com.google.cloud.tools.managedcloudsdk.MessageListener;
+import com.google.cloud.tools.managedcloudsdk.process.AsyncStreamHandler;
+import com.google.cloud.tools.managedcloudsdk.process.CommandExecutor;
+import com.google.cloud.tools.managedcloudsdk.process.CommandExecutorFactory;
+import com.google.common.base.Joiner;
+import java.io.IOException;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.ExecutionException;
+
+/** Gcloud wrapper for updating the Cloud SDK. */
+final class Updater {
+
+  private final Path gcloud;
+  private final MessageListener messageListener;
+  private final CommandExecutorFactory commandExecutorFactory;
+  private final AsyncStreamHandler<Void> stdOutListener;
+  private final AsyncStreamHandler<Void> stdErrListener;
+
+  /** Instantiated by {@link UpdaterFactory}. */
+  Updater(
```

Comment:
Might make sense to have this share an abstract parent with ComponentInstaller, maybe like GcloudWrapper?